rust/lazy/src/lazy_rule.rs

```diff
@@ -0,0 +1,142 @@
+use std::{path::Path, str::FromStr};
+
+use crate::error::{LazyError, Result};
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum LazyStrategy {
+    SMUDGE,
+    POINTER,
+}
+
+impl FromStr for LazyStrategy {
+    type Err = LazyError;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let lower_s = s.to_ascii_lowercase();
+        match lower_s.as_str() {
+            "smudge" => Ok(LazyStrategy::SMUDGE),
+            "pointer" => Ok(LazyStrategy::POINTER),
+            "s" => Ok(LazyStrategy::SMUDGE),
+            "p" => Ok(LazyStrategy::POINTER),
+            _ => Err(LazyError::InvalidStrategy(format!(
+                "{s}, supported strategy is smudge(s) or pointer(p)"
+            ))),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct LazyRule {
+    pub path: String,
+    pub strategy: LazyStrategy,
+}
+
+impl FromStr for LazyRule {
+    type Err = LazyError;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let split = s.trim().split_once(' ').ok_or_else(|| {
```

Comment:
Instead of splitting on just space, I would use something that allows splitting on tabs as well: 
```
let s = s.trim();
let Some(n) = s.find(|c| c.is_whitespace()) else { return LazyError::InvalidRule(format!("Failed to parse line {s:?}")); };
let (rule, path) = s.split_at(n);
let path = path.trim(); 
```

Reply:
Good point, I can change to `split_whitespace`

Reply:
Actually, `split_once(char::is_whitespace)`